salto-io/salto
# SALTO-4568:[zendesk] Add change validator to make sure no deactivated ticket forms are used in view conditions

This change validator is meant to check that when a `view` is modified, it doesn't use a deactivated `ticket form` in its conditions. This also catches cases where the `ticket form` used is being deactivated in the same modification.

Also included is a test for this change validator

---

Manual tests made: 
1. Fetch, change view, deploy -> no error
2. Change view to include a deactivated ticket form -> error shows up
3. Change view to include an active ticket form -> no error
4. Change multiple views to include multiple deactivated ticket forms -> multiple errors show up
5. Change views to include a ticket form and also change that ticket form to be inactive -> error shows up

---
_Release Notes_: 

Zendesk: 
When trying to deploy a change to a `view` that has a condition with a reference to a deactivated `ticket form`, an error will appear.
